pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -201,9 +201,6 @@ type PriorityQueue struct {
 	metricsRecorder metrics.MetricAsyncRecorder
 	// pluginMetricsSamplePercent is the percentage of plugin metrics to be sampled.
 	pluginMetricsSamplePercent int
-
-	// isSchedulingQueueHintEnabled indicates whether the feature gate for the scheduling queue is enabled.
-	isSchedulingQueueHintEnabled bool
```

Comment:
follow up: https://github.com/kubernetes/kubernetes/pull/118438#discussion_r1265806317

Reply:
I think my comment was misinterpreted.

It is ok to have a proxy boolean for the feature gate (in fact, that's the only way to write unit-tests that can run in parallel).

What is not ok is not to guard a piece of code with a feature gate check based on the assumption that another section is not buggy.

Reply:
Understood. 🙏